lib/Sema/MiscDiagnostics.cpp

```diff
@@ -2992,13 +2992,19 @@ static void checkStmtConditionTrailingClosure(TypeChecker &TC, const Expr *E) {
     bool shouldWalkIntoNonSingleExpressionClosure() override { return false; }
 
     std::pair<bool, Expr *> walkToExprPre(Expr *E) override {
-      // Dig into implicit expression.
-      if (E->isImplicit()) return { true, E };
-      // Diagnose call expression.
-      if (auto CE = dyn_cast<CallExpr>(E))
+      // If a trailing closure appears as a child of one of these types of
+      // expression, don't diagnose it as there is no ambiguity
+      static const std::set<ExprKind> skippedExpressionKinds{
+          ExprKind::Paren, ExprKind::Tuple, ExprKind::Array,
+          ExprKind::Dictionary, ExprKind::InterpolatedStringLiteral};
```

Comment:
Instead of using `std::set`, please use `switch` for this.
```cpp
switch (E->getKind()) {
case ExprKind::Paren:
case ExprKind::Tuple:
case ExprKind::Array:
case ExprKind::Dictionary:
case ExprKind::InterpolatedStringLiteral:
  // If a trailing closure appears as a child of one of these types of
  // expression, don't diagnose it as there is no ambiguity.	
  return { E->isImplicit(), E };
case ExprKind::Call
  diagnoseIt(CE)
  break;
default;
  break;
}
return { true, E };
```